src/components/QueryEditor/VisualQueryEditor/utils/utils.ts

```diff
@@ -195,3 +197,14 @@ export function sanitizeGroupBy(expression: QueryEditorGroupByExpression): Query
 
   return undefined;
 }
+
+export function filterColumns(
+  tableSchema?: AdxColumnSchema[],
+  expression?: QueryEditorColumnsExpression
+): AdxColumnSchema[] | undefined {
+  return expression?.columns?.length
+    ? // filter columns with the same name or under the same dynamic column
+      // e.g. MyCol or MyCol["Inner"]
+      tableSchema?.filter((c) => expression?.columns?.includes(c.Name.split('[')[0]))
```

Comment:
I can't figure out a clearer way to express this without looping twice :sweat_smile:, nice work!